### ISSUE TITLE
<Notification /> - Warning in React 16

### ISSUE THREAD

**Issue by rghorbani** [wix/wix-style-react#2523]
<Notification produces a warning even when all the conditions are met.
<img width="1073" alt="screen shot 2018-12-02 at 9 16 33 am" src="https://user-images.githubusercontent.com/5933939/49336224-7276cd00-f613-11e8-9879-e95537af74d6.png">

```
<Notification
  show={this.state.notificationVisible}
  theme={this.state.notificationTheme}
  type="sticky"
  timeout={2000}
  zIndex={10000}
  onClose={() =>
    this._isMounted &&
    this.setState({ notificationVisible: false })
  }
>
  <Notification.TextLabel>
    {this.state.notificationMessage}
  </Notification.TextLabel>

  <Notification.CloseButton />
</Notification>
```

# Bug Report

### Relevant Components
<Notification />

### Current Behavior
Behavior is ok but produces a warning

### Severity
Low

**Comment by shlomitc** [comment]
Can you verify it is actually React 16 related and not something related to your project configuration?
Maybe try it with React 15 just to make sure

**Comment by rghorbani** [comment]
@shlomitc 
This is my code
```
<Notification
  show={config.notificationVisible}
  theme={config.notificationTheme}
  type="sticky"
  timeout={2000}
  zIndex={10000}
  onClose={() =>
    this.props.dispatch(hideNotification())
  }
>
  <Notification.TextLabel>
    {config.notificationMessage}
  </Notification.TextLabel>

  <Notification.CloseButton />
</Notification>
```

**Comment by hepiyellow** [comment]
There was a change with CloseButton yesterday, maybe it doesnt pass the prop validation by type now.

**Comment by shlomitc** [comment]
@mykas - can you please check if it's related to the recent `<CloseButton/>` changes?

**Comment by mykas** [comment]
This issue is 3 days ago. I have only released it yesterday

**Comment by mykas** [comment]
The error says that the order of childrens is wrong. So it might be something internal.

**Comment by shlomitc** [comment]
done and released in 5.16.0
